src/main/java/decodex/Decodex.java

```diff
@@ -55,46 +54,22 @@ public static void main(String[] args) {
     }
 
     public void run() {
+        ui.printGreeting();
+
         Command command = null;
 
         do {
-            printPromptHeader();
-            String userInput = in.nextLine();
+            String userInput = ui.readInput(in);
```

Comment:
Since the `Ui` object has already been instantiated, perhaps you could let `Ui` to hold the `Scanner`?